pkg/cmd/cli/cmd/set/env.go

```diff
@@ -48,6 +48,12 @@ syntax.`
   # Update all containers in all replication controllers in the project to have ENV=prod
   %[1]s env rc --all ENV=prod
 
+  # Import environment from a secret
+  %[1]s env --from-secret=mysecret dc/myapp
```

Comment:
My concern is that we don't know yet whether we'd have other flags because
no one has gone through the resources and said "this makes sense" or "this
would be crazy".  Someone doing that would be good.

On Tue, Jun 7, 2016 at 11:24 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/env.go
> <https://github.com/openshift/origin/pull/9210#discussion_r66187732>:
>
> > @@ -48,6 +48,12 @@ syntax.`
> >    # Update all containers in all replication controllers in the project to have ENV=prod
> >    %[1]s env rc --all ENV=prod
> >
> > +  # Import environment from a secret
> > +  %[1]s env --from-secret=mysecret dc/myapp
>
> there's nothing that prevents the client from still having
> resource-specific intelligence even if the syntax is
> --from=secret/mysecret, since the client still knows the resource type. so
> i would also vote for that approach, unless we think we're also going to be
> introducing additional arguments that only make sense for one type or
> another (at which point yes, saying "you can't use --from-secret with
> --configflag" is nicer than "you can't use --from with --configflag because
> you passed a secret resource type")
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9210/files/2dca495d9d658651af8fa61da45494ba31c65f6f#r66187732>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p_tESMzfK1MTJJVFQPiFwyXaggJ-ks5qJjYFgaJpZM4IwT6g>
> .
>